src/ucp/core/ucp_ep.c

```diff
@@ -785,8 +796,24 @@ void ucp_ep_disconnected(ucp_ep_h ep, int force)
         return;
     }
 
-    ucp_ep_match_remove_ep(ep->worker, ep);
+    ucp_ep_match_remove_ep(worker, ep);
     ucp_ep_destroy_internal(ep);
+
+    /* remove pending slow-path functions after EP cleanup, because it does
+     * cleanup lanes which purges all outstanding operation and purged
+     * operations coudl add callbacks on the progress */
```

Comment:
but can't these callbacks try to access a destroyed ep?
which purged operation can add calllacks?

Reply:
they don't touch a EP